src/main/scala/mesosphere/marathon/state/TaskFailure.scala

```diff
@@ -65,6 +66,26 @@ object TaskFailure {
       slaveId = if (proto.hasSlaveId) Some(proto.getSlaveId) else None
     )
 
+  object FromUnhealthyTaskKillEvent {
+    def unapply(event: UnhealthyTaskKillEvent): Option[TaskFailure] =
+      apply(event)
+
+    def apply(event: UnhealthyTaskKillEvent): Option[TaskFailure] = {
+      val UnhealthyTaskKillEvent(appId, taskId, version, health, host, slaveID, _, _) = event
+
+      Some(TaskFailure(
+        appId,
+        taskId.mesosTaskId,
+        mesos.TaskState.TASK_KILLED,
+        health.lastFailureCause.getOrElse(""),
```

Comment:
Perhaps we should always prepend something like:
`"Task was killed since health check failed. Reason: $reason"`
Also: if the failureCause is unset: `getOrElse("unknown")` would be better than an empty string.